rcgen.js

```diff
@@ -0,0 +1,88 @@
+// @ts-check
+
+const {gitFiles} = require('./rcgen/git/files');
+const {ignoreFilesWithGit} = require('./rcgen/git/utils');
+const {prettierFiles} = require('./rcgen/prettier/files');
+const {prettierPatchers} = require('./rcgen/prettier/patchers');
+const {
+  configurePrettier,
+  ignoreFilesWithPrettier
+} = require('./rcgen/prettier/utils');
+const {vscodeFiles} = require('./rcgen/vscode/files');
+const {vscodePatchers} = require('./rcgen/vscode/patchers');
+const {
+  excludeFilesFromSearchInVscode,
+  excludeFilesInVscode,
+  recommendVscodeExtensions
+} = require('./rcgen/vscode/utils');
+
+/**
+ * @type {import('@rcgen/core').Manifest}
+ */
+exports.default = {
+  files: [...gitFiles, ...prettierFiles, ...vscodeFiles],
+  patchers: [
+    ...prettierPatchers,
+    ...vscodePatchers,
+    ignoreFilesWithGit(
+      '.cache',
+      'coverage',
+      'lerna-debug.log',
+      'lib',
+      'node_modules',
+      'npm-debug.log',
+      'package-lock.json',
+      'packages/website/build',
+      'packages/website/i18n',
+      'todo.tasks',
+      'yarn-error.log'
+    ),
+    configurePrettier({
+      bracketSpacing: false,
+      proseWrap: 'always',
+      singleQuote: true
+    }),
+    ignoreFilesWithPrettier(
+      '.cache',
+      'CHANGELOG.md',
+      'coverage',
+      'lib',
+      'node_modules',
+      'package.json',
```

Comment:
isn't there a conceptual problem here? I imagine that you envision this line to be contributed by the `npm` plugin. But isn't this an `n`-`m` problem starting to arise? Because now all plugins for tools which write files need to know of all formatters people could use for this filetype...